File: jambeth/jAmbeth.Filter/src/de/osthus/ambeth/filter/model/SortDescriptor.java
package de.osthus.ambeth.filter.model;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlType;

@XmlType
@XmlAccessorType(XmlAccessType.FIELD)
public class SortDescriptor implements ISortDescriptor
{
	@XmlElement(name = "Member", required = true)
	protected String member;

	@XmlElement(name = "SortDirection", required = true)
	protected SortDirection sortDirection = SortDirection.ASCENDING;

	@Override
	public String getMember()
	{
		return member;
	}

	public void setMember(String member)
	{
		this.member = member;
	}

	public SortDescriptor withMember(String member)
	{
		setMember(member);
		return this;
	}

	@Override
	public SortDirection getSortDirection()
	{
		return sortDirection;
	}

	public void setSortDirection(SortDirection sortDirection)
	{
		this.sortDirection = sortDirection;
	}

	public SortDescriptor withSortDirection(SortDirection sortDirection)
	{
		setSortDirection(sortDirection);
		return this;
	}
}

File: ambeth/Ambeth.Filter/ambeth/filter/model/SortDescriptor.cs
using System;
using De.Osthus.Ambeth.Annotation;
using System.Runtime.Serialization;
using System.Reflection;

namespace De.Osthus.Ambeth.Filter.Model
{
    [DataContract(Name = "SortDescriptor", Namespace = "http://schemas.osthus.de/Ambeth")]
    public class SortDescriptor : ISortDescriptor
    {
        [DataMember]
        public String Member { get; set; }

        [DataMember]
        public SortDirection SortDirection { get; set; }

        public SortDescriptor()
        {
            SortDirection = SortDirection.ASCENDING;
        }

        public SortDescriptor WithMember(String member)
        {
            Member = member;
            return this;
        }

        public SortDescriptor WithSortDirection(SortDirection sortDirection)
        {
            SortDirection = sortDirection;
            return this;
        }

        public override bool Equals(Object obj)
        {
            if (Object.ReferenceEquals(this, obj))
            {
                return true;
            }
            if (!(obj is SortDescriptor))
            {
                return false;
            }
            SortDescriptor compared = (SortDescriptor)obj;
            if (!Object.Equals(Member, compared.Member))
            {
                return false;
            }
            if (!Object.Equals(SortDirection, compared.SortDirection))
            {
                return false;
            }
            return true;
        }

        public override int GetHashCode()
        {
            return (Member.GetHashCode() ^ SortDirection.GetHashCode());
        }
    }
}